core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -235,13 +230,14 @@ private[spark] class AppStatusListener(
               rdd.diskUsed = addDeltaToValue(rdd.diskUsed,
                 (partition.diskUsed / partition.executors.length) * -1)
               partition.update(partition.executors
-                .filter(!_.equals(event.executorId)), rdd.storageLevel,
+                .filter(!_.equals(event.executorId)),
                 addDeltaToValue(partition.memoryUsed,
                   (partition.memoryUsed / partition.executors.length) * -1),
                 addDeltaToValue(partition.diskUsed,
                   (partition.diskUsed / partition.executors.length) * -1))
             }
           }
+        }
         update(rdd, now)
```

Comment:
Hmm, not sure where you're seeing the conditional update? This line is executed for all live RDDs in the old code.

Reply:
I guess the `update` in this line was unnecessary? 

https://github.com/apache/spark/blob/4c8f114783241e780c0ad227bfb3191657637df7/core/src/main/scala/org/apache/spark/status/AppStatusListener.scala#L219-L221

I think this is the `update` which caused the confusion for @squito too.

Reply:
right, it was that part, combined with misreading this part of the old code.  Though it seems like it could be a conditional update, but no change from before.